utils/pythonutils/piputils.go

```diff
@@ -0,0 +1,220 @@
+		if err != nil {
+			return "", err
+		}
+		dependenciesDirName = filepath.Join(home, dependenciesDirName, "pip")
+	}
+	depTreeScriptName := "pipdeptree.py"
+	pipDependenciesPath := filepath.Join(dependenciesDirName, "pip", pipDepTreeVersion)
+	depTreeScriptPath := filepath.Join(pipDependenciesPath, depTreeScriptName)
+	err := writeScriptIfNeeded(pipDependenciesPath, depTreeScriptName)
+	if err != nil {
+		return "", err
+	}
+	return depTreeScriptPath, err
+}
+
+// Creates local python script on jfrog dependencies path folder if such not exists
+func writeScriptIfNeeded(targetDirPath, scriptName string) error {
+	scriptPath := filepath.Join(targetDirPath, scriptName)
+	exists, err := utils.IsFileExists(scriptPath, false)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		err = os.MkdirAll(targetDirPath, os.ModeDir|os.ModePerm)
+		if err != nil {
+			return err
+		}
+		err = ioutil.WriteFile(scriptPath, pipDepTreeContent, os.ModePerm)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func GetPackageNameFromSetuppy(srcPath string) (string, error) {
+	filePath, err := getSetupPyFilePath(srcPath)
+	if err != nil || filePath == "" {
+		// Error was returned or setup.py does not exist in directory.
+		return "", err
+	}
+
+	// Extract package name from setup.py.
+	packageName, err := ExtractPackageNameFromSetupPy(filePath)
+	if err != nil {
+		// If setup.py egg_info command failed we use build name as module name and continue to pip-install execution
+		return "", errors.New("Couldn't determine module-name after running the 'egg_info' command: " + err.Error())
+	}
+	return packageName, nil
+}
+
+// Look for 'setup.py' file in current work dir.
+// If found, return its absolute path.
+func getSetupPyFilePath(srcPath string) (string, error) {
+	filePath := filepath.Join(srcPath, "setup.py")
+	// Check if setup.py exists.
+	validPath, err := utils.IsFileExists(filePath, false)
+	if err != nil {
+		return "", err
+	}
+	if !validPath {
+		return "", nil
+	}
+
+	return filePath, nil
+}
+
+// Get the project-name by running 'egg_info' command on setup.py and extracting it from 'PKG-INFO' file.
+func ExtractPackageNameFromSetupPy(setuppyFilePath string) (string, error) {
+	// Execute egg_info command and return PKG-INFO content.
+	content, err := getEgginfoPkginfoContent(setuppyFilePath)
+	if err != nil {
+		return "", err
+	}
+
+	// Extract project name from file content.
+	return getProjectIdFromFileContent(content)
+}
+
+// Run egg-info command on setup.py, the command generates metadata files.
+// Return the content of the 'PKG-INFO' file.
+func getEgginfoPkginfoContent(setuppyFilePath string) (output []byte, err error) {
+	eggBase, err := utils.CreateTempDir()
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		e := utils.RemoveTempDir(eggBase)
+		if err == nil {
+			err = e
+		}
+	}()
+
+	// Run python 'egg_info --egg-base <eggBase>' command.
+	pythonExecutablePath, err := exec.LookPath("python")
```

Comment:
The above line finds the path fpr the python executable. However, I noticed that our code also include the following logic - 
```	pythonExecutable, pathErr := exec.LookPath("python3")
	if pathErr != nil || pythonExecutable == "" {
		if runtime.GOOS == "windows" {
			// If the OS is Windows try using Py Launcher: 'py -3'
			pythonExecutable, pathErr = exec.LookPath("py")
			if pathErr != nil && pythonExecutable != "" {
				cmdName = "-3"
			}
		}
		// Try using 'python' if 'python3'/'py' couldn't been found
		if pythonExecutable == "" {
			pythonExecutable = "python"
		}
	}
```
Doesn't it apply here as well? If so, worth creating a ```LookPythonPath``` utility to do this, and reuse it everywhere.